WordPress/Classes/ViewRelated/Reader/ReaderDetailViewController.swift

```diff
@@ -0,0 +1,903 @@
+        richTextView.refreshMediaLayout()
+    }
+
+
+    // MARK: - Setup
+
+    public func setupWithPostID(postID:NSNumber, siteID:NSNumber) {
+        let title = NSLocalizedString("Loading Post...", comment:"Text displayed while loading a post.")
+        WPNoResultsView.displayAnimatedBoxWithTitle(title, message: nil, view: view)
+
+        let context = ContextManager.sharedInstance().mainContext
+        let service = ReaderPostService(managedObjectContext: context)
+
+        service.fetchPost(
+            postID.unsignedIntegerValue,
+            forSite: siteID.unsignedIntegerValue,
+            success: {[weak self] (post:ReaderPost!) in
+                self?.post = post
+                WPNoResultsView.removeFromView(self?.view)
+            },
+            failure: {[weak self] (error:NSError!) in
+                DDLogSwift.logError("Error fetching post for detail: \(error.localizedDescription)")
+
+                let title = NSLocalizedString("Error Loading Post", comment:"Text displayed when load post fails.")
+                WPNoResultsView.displayAnimatedBoxWithTitle(title, message: nil, view: self?.view)
+            }
+        )
+    }
+
+
+    private func setupNavBar() {
+        configureNavTitle()
+
+        // Don't show 'Reader' in the next-view back button
+        navigationItem.backBarButtonItem = UIBarButtonItem(title: " ", style: .Plain, target: nil, action: nil)
+    }
+
+
+    // MARK: - Configuration
+
+    /**
+    Applies the default styles to the cell's subviews
+    */
+    private func applyStyles() {
+        WPStyleGuide.applyReaderCardSiteButtonStyle(blogNameButton)
+        WPStyleGuide.applyReaderCardBylineLabelStyle(bylineLabel)
+        WPStyleGuide.applyReaderCardBylineLabelStyle(blogURLLabel)
+        WPStyleGuide.applyReaderCardTitleLabelStyle(titleLabel)
+        WPStyleGuide.applyReaderCardTagButtonStyle(tagButton)
+        WPStyleGuide.applyReaderCardActionButtonStyle(commentButton)
+        WPStyleGuide.applyReaderCardActionButtonStyle(likeButton)
+    }
+
+
+    private func configureView() {
+        scrollView.alpha = 1
+        configureNavTitle()
+        configureShareButton()
+        configureHeader()
+        configureFeaturedImage()
+        configureTitle()
+        configureByLine()
+        configureRichText()
+        configureDiscoverAttribution()
+        configureTag()
+        configureActionButtons()
+        configureFooterIfNeeded()
+
+        bumpStats()
+        bumpPageViewsForPost()
+
+        NSNotificationCenter.defaultCenter().addObserver(self,
+            selector: "handleBlockSiteNotification:",
+            name: ReaderPostMenu.BlockSiteNotification,
+            object: nil)
+    }
+
+
+    private func configureNavTitle() {
+        if let postTitle = post?.postTitle {
+            self.title = postTitle
+        } else {
+            self.title = NSLocalizedString("Post", comment:"Placeholder title for ReaderPostDetails.")
+        }
```

Comment:
Actually, I like that approach better (its more concise), and if the placeholder string is assigned to its own variable (`let placeholder = ...`) then it would keep things very readable.  Let's do that.